src/target/nrf54l.c

```diff
@@ -0,0 +1,189 @@
+#include "general.h"
+#include "target.h"
+#include "target_internal.h"
+#include "cortexm.h"
+#include "adiv5.h"
+
+#define NRF54L_PARTNO 0x1c0U
+
+#define NRF54L_FICR_INFO_RAM  0x00ffc328U
+#define NRF54L_FICR_INFO_RRAM 0x00ffc32cU
+
+#define NRF54L_RRAM 0x00000000U
+#define NRF54L_UICR 0x00ffd000U
+#define NRF54L_RAM  0x20000000U
+
+#define NRF54L_RRAMC_READY                   0x5004b400U
+#define NRF54L_RRAMC_BUFSTATUS_READYNEXT     0x5004b404U
+#define NRF54L_RRAMC_BUFSTATUS_WRITEBUFEMPTY 0x5004b418U
+#define NRF54L_RRAMC_CONFIG                  0x5004b500U
+#define NRF54L_RRAMC_ERASE_ERASEALL          0x5004b540U
+
+#define NRF54L_RRAMC_CONFIG_WEN                (1U << 0U)
+#define NRF54L_RRAMC_CONFIG_WRITEBUFSIZE(size) (size << 8U)
+
+#define NRF54L_CTRL_AP_IDR_VALUE 0x32880000U
+
+#define NRF54L_CTRL_AP_RESET            ADIV5_AP_REG(0x00U)
+#define NRF54L_CTRL_AP_ERASEALL         ADIV5_AP_REG(0x04U)
+#define NRF54L_CTRL_AP_ERASEALLSTATUS   ADIV5_AP_REG(0x08U)
+#define NRF54L_CTRL_AP_APPROTECT_STATUS ADIV5_AP_REG(0x14U)
+
+#define NRF54L_CTRL_AP_RESET_NORESET                            0U
+#define NRF54L_CTRL_AP_RESET_HARDRESET                          2U
+#define NRF54L_CTRL_AP_ERASEALLSTATUS_BUSY                      2U
+#define NRF54L_CTRL_AP_APPROTECT_STATUS_APPROTECT_ENABLED       (1U << 0U)
+#define NRF54L_CTRL_AP_APPROTECT_STATUS_SECUREAPPROTECT_ENABLED (1U << 1U)
+
+static bool rram_erase(target_flash_s *flash, target_addr_t addr, size_t len)
+{
+	(void)flash;
+	(void)addr;
+	(void)len;
+	// RRAM doesn't need to be erased before being written, so we just return ok.
+	return true;
+}
+
+static bool rram_write(target_flash_s *flash, target_addr_t dest, const void *src, size_t len)
+{
+	while (target_mem32_read32(flash->t, NRF54L_RRAMC_BUFSTATUS_READYNEXT) != 1U) {
+		// Wait for rram to be ready for next write.
+	}
```

Comment:
Please drop the braces in preference for `continue;` as the loop body as is the style in the rest of the code base. eg:

```suggestion
	// Wait for RRAM to be ready for next write.
	while (target_mem32_read32(flash->t, NRF54L_RRAMC_BUFSTATUS_READYNEXT) != 1U)
		continue;
```

It would be useful to include a `#define` above for what the magic number being checked means.

Same request for `rram_done()` which has the same pattern but for a different register.